types/colors.ts

```diff
@@ -0,0 +1,30 @@
+type IntentOptions = 'primary' | 'secondary' | 'neutral' | 'success' | 'danger' | 'info';
+type ModeOptions = 'default' | 'dark';
+type ProminenceOptions = 'default' | 'spicy' | 'mild';
+type UsageOptions = 'background' | 'border' | 'icon' | 'text';
+export type InteractionOptions = 'hover' | 'active' | 'disabled';
+
+export type ColorInteractionType = { interaction: InteractionOptions };
+export type ColorProminenceType = { prominence: ProminenceOptions };
+export type ColorUsageType = { usage: UsageOptions };
+export type ColorModeType = { mode: ModeOptions };
+export type ColorIntentType = { intent: IntentOptions };
+
+export type CompoundVariantType = {
```

Comment:
This is just my lack of experience here, but is compound variant a standard in some way or something you've come up with? Either way, seems like something good to document